src/modules.js

```diff
@@ -160,4 +184,105 @@ if (__analytics.amplitude.isEnabled) {
 }
 
 
+// --- Infermedica Analytics ---
+if(__analytics.infermedicaAnalytics?.isEnabled) {
+  const infermedicaModule = function () {
+    const hasAuthentication = false;
+    const baseURL = __analytics.infermedicaAnalytics?.baseURL || 'https://analytics-proxy.test.infermedica.com/';
+    const {topic, environment, firebaseConfig} = __analytics.infermedicaAnalytics;
+    const browser = Bowser.getParser(window.navigator.userAgent);
+    const analyticsApi = axios.create({
+      baseURL,
+    });
+    const publish = async function(data) {
+      const publishURL = '/api/v1/publish';
+      const attributes = {
+        environment,
+      }
+      const events = [
+        {
+          data
+        }
+      ];
+      const payload = JSON.stringify({
+        topic,
+        events,
+      });
+
+      await analyticsApi.post(publishURL, payload);
+    }
+
+    const firebaseData = {};
+    firebase.initializeApp(firebaseConfig);
+    (async function () {
+      try {
+        await firebase.auth().signInAnonymously();
+      }
+      catch (error) {
+        console.error(error);
+      }
+    })();
+    const eventQueue = [];
+    firebase.auth().onAuthStateChanged(async (user) => {
+      if(!user) return;
+      firebaseData.uid = user.uid;
+      firebaseData.token = await user.getIdToken();
+      if(hasAuthentication) {
+        analyticsApi.defaults.headers.Authorization = `Bearer ${firebaseData.token}`
+      }
+      eventQueue.forEach((event) => {
+        const {user} = event;
+        publish({ ...event, user: {...user, id: firebaseData.uid }});
```

Comment:
H: It would be good to removed the messages form `eventQueue` after sending them.